api/src/main/java/org/pmiops/workbench/cohorts/CohortService.java

```diff
@@ -1,54 +1,78 @@
 package org.pmiops.workbench.cohorts;
 
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.storage.Blob;
 import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Streams;
+import com.google.gson.Gson;
+import java.nio.charset.StandardCharsets;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
 import java.util.stream.Collectors;
 import javax.inject.Provider;
+import org.pmiops.workbench.api.BigQueryService;
+import org.pmiops.workbench.cohortbuilder.CohortQueryBuilder;
+import org.pmiops.workbench.cohortbuilder.ParticipantCriteria;
 import org.pmiops.workbench.config.WorkbenchConfig;
 import org.pmiops.workbench.db.dao.CohortDao;
 import org.pmiops.workbench.db.model.DbCohort;
+import org.pmiops.workbench.db.model.DbWorkspace;
 import org.pmiops.workbench.firecloud.ApiException;
+import org.pmiops.workbench.firecloud.FireCloudService;
 import org.pmiops.workbench.firecloud.api.MethodConfigurationsApi;
 import org.pmiops.workbench.firecloud.api.SubmissionsApi;
 import org.pmiops.workbench.firecloud.model.FirecloudMethodConfiguration;
 import org.pmiops.workbench.firecloud.model.FirecloudSubmissionRequest;
 import org.pmiops.workbench.firecloud.model.FirecloudSubmissionResponse;
+import org.pmiops.workbench.firecloud.model.FirecloudWorkspace;
+import org.pmiops.workbench.google.CloudStorageClient;
+import org.pmiops.workbench.google.StorageConfig;
 import org.pmiops.workbench.model.Cohort;
+import org.pmiops.workbench.model.SearchRequest;
 import org.pmiops.workbench.model.TerraJob;
 import org.pmiops.workbench.model.TerraJobStatus;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Service;
 
 @Service
 public class CohortService {
 
+  private final BigQueryService bigQueryService;
+  private final CohortQueryBuilder cohortQueryBuilder;
   private final CohortDao cohortDao;
   private final CohortMapper cohortMapper;
+  private final Provider<CloudStorageClient> cloudStorageClientProvider;
+  private final FireCloudService fireCloudService;
   private final Provider<SubmissionsApi> submissionApiProvider;
   private final Provider<MethodConfigurationsApi> methodConfigurationsApiProvider;
   private final Provider<WorkbenchConfig> workbenchConfigProvider;
 
   @Autowired
   public CohortService(
+      BigQueryService bigQueryService,
+      CohortQueryBuilder cohortQueryBuilder,
       CohortDao cohortDao,
       CohortMapper cohortMapper,
+      @Qualifier(StorageConfig.WGS_EXTRACTION_STORAGE_CLIENT)
```

Comment:
What's going to happen when we push this to staging, stable, etc? Should confirm that an empty SA still works properly